osu.Game/Screens/Backgrounds/BackgroundModeBeatmap.cs

```diff
@@ -22,17 +22,19 @@ public WorkingBeatmap Beatmap
             {
                 return beatmap;
             }
-
             set
             {
-                if (beatmap == value)
+                if (beatmap == value && beatmap != null)
                     return;
-
                 beatmap = value;
 
                 Schedule(() =>
                 {
-                    Background newBackground = new BeatmapBackground(beatmap);
+                    Background newBackground;
+                    if (beatmap == null)

```

Comment:
Checking if `beatmap == Default` is the same thing as `beatmap == null`, only sillier.

The beatmap still needs to be imported into the database and written into the filesystem, which takes time. If you don't want to write it in the filesystem, the maintenance cost and complexity overhead of having the beatmap store support some kind of virtual beatmap that's not sourced from the filesystem is nuts compared to just checking if there are no beatmaps in the db.